From 6058a641d32433c69afd66f4c40ba184ed7b735c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Thu, 4 Jul 2013 22:59:33 +0200
Subject: wined3d: Get rid of the end_scene flush and finish

Either keep the patch this way, or send a flush through the CS. I'm not
sure it's worth it though, this has never had any real performance
impact.
---
 dlls/wined3d/device.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/dlls/wined3d/device.c b/dlls/wined3d/device.c
index d851c52..3d17a63 100644
--- a/dlls/wined3d/device.c
+++ b/dlls/wined3d/device.c
@@ -3329,8 +3329,6 @@ HRESULT CDECL wined3d_device_begin_scene(struct wined3d_device *device)
 
 HRESULT CDECL wined3d_device_end_scene(struct wined3d_device *device)
 {
-    struct wined3d_context *context;
-
     TRACE("device %p.\n", device);
 
     if (!device->inScene)
@@ -3339,16 +3337,6 @@ HRESULT CDECL wined3d_device_end_scene(struct wined3d_device *device)
         return WINED3DERR_INVALIDCALL;
     }
 
-    context = context_acquire(device, NULL);
-    /* We only have to do this if we need to read the, swapbuffers performs a flush for us */
-    if (wined3d_settings.cs_multithreaded)
-        context->gl_info->gl_ops.gl.p_glFinish();
-    else
-        context->gl_info->gl_ops.gl.p_glFlush();
-    /* No checkGLcall here to avoid locking the lock just for checking a call that hardly ever
-     * fails. */
-    context_release(context);
-
     device->inScene = FALSE;
     return WINED3D_OK;
 }
-- 
2.8.0

